packages/core/src/jobs/job-definitions/batchEvaluations/runDocumentJob.ts

```diff
@@ -59,15 +60,11 @@ export const runDocumentForEvaluationJob = async (
       source: LogSources.Evaluation,
     }).then((r) => r.unwrap())
 
-    // TODO handle this error better
-    const res = await result.response
-    if (!res) throw new BadRequestError('Document run failed')
```

Comment:
how is this handled now? runDocumentAtCommit has to always return before result.response resolves yet result.response might throw, so we need to wait for response to finish in this particular case.

Reply:
`result.response` never throws now. Please review `runChain` and tell me this is the case